internal/auth0/organization/resource_connections.go

```diff
@@ -198,11 +198,25 @@ func guardAgainstErasingUnwantedConnections(
 		return nil
 	}
 
+	alreadyEnabledConnectionsIDs := make([]string, len(alreadyEnabledConnections))
+	for _, conn := range alreadyEnabledConnections {
+		alreadyEnabledConnectionsIDs = append(alreadyEnabledConnectionsIDs, conn.GetConnectionID())
+	}
+
+	connectionIDsToAdd := make([]string, len(connectionsToAdd))
+	for _, conn := range connectionsToAdd {
+		connectionIDsToAdd = append(connectionIDsToAdd, conn.GetConnectionID())
+	}
+
+	if cmp.Equal(connectionIDsToAdd, alreadyEnabledConnectionsIDs) {
+		return nil
+	}
```

Comment:
While the two objects are of the same type, the organization connections retrieves by the API contain a few additional details that will make the diff always fail. Considering we only care about the diff between connection IDs we loop through both objects and only collect those.